### PR TITLE
fix(grid): Selecting only filtered rows #6260

### LINKED ISSUE #6260: Row selectors headers are not updated correct when filter with Advanced Filter
## Description  
Row selectors headers are not updated correct when filter with Advanced Filter

 * igniteui-angular version: 9.0
 * browser: n/a

## Steps to reproduce  
Scenario 1: 
1. Open dev sample gridFiltering
2. Click on header checkbox to select all rows
3. Clear advanced filter

Scenario 2: 
1. Open dev sample gridFiltering
2. Clear advanced filter
3. Select first row
4. Open advanced filter and filter column Id with condition 'Starts with' and value 'b'

## Result  
In Scenario 1:
All rows are selected

In Scenario 2:
Header row checkbox is partial checked

## Expected result  
In Scenario 1:
Only 3 rows should be selected and the other should not be selected

In Scenario 2:
Header row checkbox should be unchecked

## Attachments  
image

### PR DESCRIPTION
**Pull request by astaev**
Closes #6260 

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/core/grid-selection.ts [hunk 1]
```diff
@@ -688,7 +688,9 @@ export class IgxGridSelectionService {
     }
 
     private isFilteringApplied(): boolean {
-        return this.grid.filteringExpressionsTree.filteringOperands.length > 0;
+        const grid = this.grid as IgxGridBaseComponent;
+        return grid.filteringExpressionsTree.filteringOperands.length > 0 ||
+        grid.advancedFilteringExpressionsTree && grid.advancedFilteringExpressionsTree.filteringOperands.length > 0;
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/core/grid-selection.ts]: You could use the static helper method `FilteringExpressionsTree.empty()`.

File: projects/igniteui-angular/src/lib/grids/grid/grid-row-selection.spec.ts [hunk 2]
```diff
@@ -1671,6 +1673,40 @@ describe('IgxGrid - Row Selection #grid', () => {
             expect(grid.getRowByIndex(1).selected).toBeTruthy();
             expect(grid.onRowSelectionChange.emit).toHaveBeenCalledTimes(6);
         }));
+
+        it('Should select only filtered records', async() => {
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/grid/grid-row-selection.spec.ts]: Try to change the test to be fakeAsync and remove the grid.cdr.detectChanges() calls if possible.

File: projects/igniteui-angular/src/lib/core/grid-selection.ts (hunk 1 above)
Comment (astaev) [reply projects/igniteui-angular/src/lib/core/grid-selection.ts]: Done

File: projects/igniteui-angular/src/lib/grids/grid/grid-row-selection.spec.ts (hunk 2 above)
Comment (astaev) [reply projects/igniteui-angular/src/lib/grids/grid/grid-row-selection.spec.ts]: Test seems to work without grid detectChanges, so just removing them.

**Pull request merged**